src/crawlee/fingerprint_suite/_browserforge_adapter.py

```diff
@@ -3,17 +3,66 @@
 from copy import deepcopy
 from typing import TYPE_CHECKING, Any
 
+from browserforge.bayesian_network import extract_json
 from browserforge.fingerprints import Fingerprint as bf_Fingerprint
 from browserforge.fingerprints import FingerprintGenerator as bf_FingerprintGenerator
 from browserforge.fingerprints import Screen
+from browserforge.headers.generator import DATA_DIR
+from browserforge.headers.generator import HeaderGenerator as bf_HeaderGenerator
 from typing_extensions import override
 
 from crawlee._utils.docs import docs_group
 
+from ._consts import BROWSER_TYPE_HEADER_KEYWORD
 from ._fingerprint_generator import FingerprintGenerator
 
 if TYPE_CHECKING:
-    from ._types import HeaderGeneratorOptions, ScreenOptions
+    from ._types import HeaderGeneratorOptions, ScreenOptions, SupportedBrowserType
+
+
+class PatchedHeaderGenerator(bf_HeaderGenerator):
+    """Browserforge `HeaderGenerator` that contains patches not accepted in upstream repo."""
+
+    def _get_accept_language_header(self, locales: tuple[str, ...]) -> str:
+        """Generates the Accept-Language header based on the given locales.
+
+        Patched version due to PR of upstream repo not being merged: https://github.com/daijro/browserforge/pull/24
+
+        Parameters:
+            locales (ListOrString): Locale(s).
+
+        Returns:
+            str: Accept-Language header string.
+        """
+        # First locale does not include quality factor, q=1 is considered as implicit.
+        additional_locales = [f'{locale};q={0.9 - index * 0.1:.1f}' for index, locale in enumerate(locales[1:])]
+        return ','.join((locales[0], *additional_locales))
+
+
+class PatchedFingerprintGenerator(bf_FingerprintGenerator):
+    """Browserforge `FingerprintGenerator` that contains patches not accepted in upstream repo."""
+
+    def __init__(  # type:ignore[no-untyped-def]  # Upstream repo types missing.
+        self,
+        *,
+        screen: Screen | None = None,
+        strict: bool = False,
+        mock_webrtc: bool = False,
+        slim: bool = False,
+        **header_kwargs,  # noqa:ANN003 # Upstream repo types missing.
+    ) -> None:
+        """Initializes the FingerprintGenerator with the given options.
```

Comment:
please just, or not to mention the class name
```suggestion
        """A default constructor.
```